containers/Connection.ts

```diff
@@ -1,54 +1,116 @@
 import { createContainer } from "unstated-next";
 import { useState, useEffect } from "react";
 import { ethers } from "ethers";
+import { API as OnboardApi, Wallet } from "bnc-onboard/dist/src/interfaces";
+import { API as NotifyApi } from "bnc-notify/dist/src/interfaces";
+import Onboard from "bnc-onboard";
+import Notify from "bnc-notify";
 import { Observable } from "rxjs";
 import { debounceTime } from "rxjs/operators";
 
 type Provider = ethers.providers.Provider;
 type Block = ethers.providers.Block;
 type Network = ethers.providers.Network;
-type ExternalProvider = ethers.providers.ExternalProvider;
 type Signer = ethers.Signer;
 
 function useConnection() {
   const [provider, setProvider] = useState<Provider | null>(null);
+  const [onboard, setOnboard] = useState<OnboardApi | null>(null);
+  const [notify, setNotify] = useState<NotifyApi | null>(null);
   const [signer, setSigner] = useState<Signer | null>(null);
   const [network, setNetwork] = useState<Network | null>(null);
   const [address, setAddress] = useState<string | null>(null);
+  const [selectedWallet, setSelectedWallet] = useState<string | null>(null);
   const [error, setError] = useState<Error | null>(null);
   const [block$, setBlock$] = useState<Observable<Block> | null>(null);
 
   const attemptConnection = async () => {
-    if (window.ethereum === undefined) {
-      throw Error("MetaMask not found, please visit https://metamask.io/");
-    }
-
-    // // for testing
-    // const provider = new ethers.providers.JsonRpcProvider()
-    // let signer = new ethers.Wallet(process.env.PRIV_KEY)
-    // signer = signer.connect(provider)
+    const apiKey = process.env.REACT_APP_ONBOARD_API_KEY
```

Comment:
For better environment maangement, I'd like to see all `process.env.VAR` refactored into their own `config.js` file. From experience, it's much easier if we have one dedicated file to detecting environment variables and/or setting default values. This way in downstream files such as this, we don't need the `VAR ? VAR : DEFAULT` pattern and can just call `VAR`.

WDYT?